libp2p/Common.h

```diff
@@ -231,7 +231,7 @@ class Node
 
     NodeID const& address() const { return id; }
 
-    explicit operator bool() const { return bool{id}; }
+    explicit operator bool() const { return (bool)id; }
```

Comment:
```suggestion
    explicit operator bool() const { return static_cast<bool>(id); }
```
This must go to a separate commit.